src/python/pants/option/config.py

```diff
@@ -192,14 +192,10 @@ def has_section(self, section):
     return self.configparser.has_section(section)
 
   def has_option(self, section, option):
-    return (self.configparser.has_option(section, option) or
-            self.configparser.has_option(self._DEFAULT_SECTION, option))
+    return self.configparser.has_option(section, option)
 
   def get_value(self, section, option):
-    if self.configparser.has_option(section, option):
-      return self.configparser.get(section, option)
-    else:
-      return self.configparser.get(self._DEFAULT_SECTION, option)
+    return self.configparser.get(section, option)
```

Comment:
AFAICT, the `DEFAULT` section is only used by `ConfigParser` for interpolation:
> The optional values can contain format strings which refer to other values in the same section, or values in a special DEFAULT section.

(from https://docs.python.org/2/library/configparser.html)

----

I believe that the only behaviour change here is the one mentioned above, which I'm happy to consider to be a bug fix.